### DISCUSSION TITLE
Extension sample -- repo for a hello world working one

### DISCUSSION THREAD

**Discussion by timheuer** [podman-desktop/podman-desktop#6714] [General]
I'm following the extension sample - https://podman-desktop.io/docs/extensions/write but the docs end at running `yarn watch` yet the package.json doesn't have any scripts for yarn to run.  Is there a repo with a better example of what the build/watch scripts should be here?

**Comment by benoitf** [comment]
you can check all samples inside the extensions folder of Podman desktop

**Comment by benoitf** [comment]
https://github.com/containers/podman-desktop/tree/main/extensions/registries this one is very light

**Reply by timheuer** [reply]
Sorry for being dense, but this still isn't working for me.

`yarn watch --extension-folder /path/to/your/extension` is failing on understanding the parameter

```
warning my-extension@0.0.1: The engine "podman-desktop" appears to be invalid.
$ vite build --watch --extension-folder .
file:///C:/Users/timheuer/AppData/Roaming/npm/node_modules/vite/dist/node/cli.js:442
          throw new CACError(`Unknown option \`${name.length > 1 ? `--${name}` : `-${name}`}\``);
                ^

CACError: Unknown option `--extensionFolder`
    at Command.checkUnknownOptions (file:///C:/Users/timheuer/AppData/Roaming/npm/node_modules/vite/dist/node/cli.js:442:17)
    at CAC.runMatchedCommand (file:///C:/Users/timheuer/AppData/Roaming/npm/node_modules/vite/dist/node/cli.js:640:13)
    at CAC.parse (file:///C:/Users/timheuer/AppData/Roaming/npm/node_modules/vite/dist/node/cli.js:579:12)
    at file:///C:/Users/timheuer/AppData/Roaming/npm/node_modules/vite/dist/node/cli.js:923:5
    at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
    at async DefaultModuleLoader.import (node:internal/modules/esm/loader:228:24)

Node.js v20.5.1
error Command failed with exit code 1.
```

no errors on npm install or anything.  Following the article and here's the package.json:

```json
{
    "devDependencies": {
        "@podman-desktop/api": "latest",
        "typescript": "latest"
    },
    "name": "my-extension",
    "displayName": "My Hello World extension",
    "description": "How to write my first extension",
    "version": "0.0.1",
    "icon": "icon.png",
    "publisher": "Tim Heuer",
    "engines": {
        "podman-desktop": "latest"
    },
    "scripts": {
        "build": "vite buildy",
        "test": "vitest run --coverage",
        "test:watch": "vitest watch --coverage",
        "watch": "vite build --watch"
    },
    "main": "./dist/extension.js",
    "contributes": {
        "commands": [
            {
                "command": "my.first.command",
                "title": "My First Extension: Hello World"
            }
        ]
    }
}
```

**Comment by benoitf** [comment]
here it's build, not buildy
```json
"build": "vite buildy"
```

then, the instruction you're using need to be run on Podman Desktop development folder

Assuming your extension is in `/home/projects/my-extension`

clone `podman-desktop` to like `/home/projects/podman-desktop`
run yarn there

then from /home/projects/podman-desktop run

`yarn watch --extension-folder /home/projects/my-extension`

**Reply by timheuer** [reply]
Got it, didn't see my typo :-) -- the last part wasn't as clear to me in the docs.  your statement of "from *podman-desktop*" is key I think -- would recommend changing that doc to add scripts and clarify this note.  Thanks for these!!!

**Reply by benoitf** [reply]
your feedback is very helpful

would you mind creating a PR against https://github.com/containers/podman-desktop/blob/main/website/docs/extensions/write/index.md with the missing pieces from a "newcomer" POV

**Reply by timheuer** [reply]
Will do!  Hitting a weird issue now that when running the podman-desktop command, it launches the desktop but this is emitted and then my hello world extension isn't there

12:34:43 PM [main] An instance of Podman Desktop is already running. Stopping
Done in 29.11s.

**Reply by benoitf** [reply]
you need to stop any running instance of Podman Desktop before issuing your command

click on the tray icon of Podman Desktop and click on `Quit`

**Reply by timheuer** [reply]
Pesky systray hiding things...
